### DISCUSSION TITLE
Craft4 : How to remove the shipping/billing address from cart from front end?

### DISCUSSION THREAD

**Discussion by aodihis** [craftcms/commerce#2797] [Q&A]
In Craft Commerce 3, we can delete the shipping or billing address from the cart by passing the billingAddressId or shippingAddressId to 0 or negative value with the autoSetNewCartAddresses config set to false, so we didn't automatically add the primary address after deletion. 

Now, how to do this scenario in Craft Commerce 4, I have try to passing 0, negative value and null to shippingAddressId/billingAddressId and shippingAddress/billingAddress and it doesn't change anything.

Or  it is not possible anymore to do deletion in Commerce4 from front end anymore?

Thank you.

**Comment by lukeholder** [comment]
The cart controller now has `clearShippingAddress` `clearBillingAddress` and `clearAddresses` params. As of https://github.com/craftcms/commerce/blob/develop/CHANGELOG.md#429---2023-05-25
